packages/bundle-source/src/types.js

```diff
@@ -1,62 +1,65 @@
+// @ts-check
+export {};
+
+/**
+ * @typedef {'endoZipBase64' | 'nestedEvaluate' | 'getExport'} ModuleFormat
+ */
+
 /**
- * @typedef {'getExport' | 'nestedEvaluate' | 'endoZipBase64'} ModuleFormat
+ * @typedef { &
+ *  BundleSourceSimple &
+ *  BundleSourceWithFormat &
+ *  BundleSourceWithOptions &
+ *  BundleSourceFallback} BundleSource
```

Comment:
It may not be obvious to reviewers, but the order of these intersections matters, insofar as Typescript treats the last one as the "most generic" version of the overloads.  That annoyed me for quite a while.

Using the generics was the best way I could find to produce the correct result type regardless of which overload you used.

Reply:
Thanks for the important note. Mind including as a `//` comment?